app/build.gradle

```diff
@@ -22,16 +24,43 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
         }
     }
+    buildFeatures {
+        dataBinding true
```

Comment:
are you using data binding? I didn't see it used anywhere

Reply:
I was going to use it with the viewmodel(that doesn't exist yet) to update the ui's directly and probably for clicklisteners as well

Reply:
gotchya - DataBinding is a pretty polarizing thing. No one seems to really be just 'okay' with it - you either hate it / love it.

I'll be curious to hear your opinion on it after a project or two with / without data binding.

Reply:
When I learned about it I had two thoughts. One, it seemed to clean up the code a lot and reduce the amount of lines that went into things, but two , I think it makes the code less readable because you can't read what's going on the main code. A lot of it is now hidden in the XML and issues could be easier to miss. I think I probably wouldn't use it much, I just thought it was the way things were going

Reply:
IMO Databinding is on its way out - the 2 things that really irk me about it is 1 like you mentioned, things hidden in XML. which i think makes it less readable, but also infinitely less testable. any logic that now lives in your XML is 100% not unit testable anymore. my second gripe with it is the build errors are pretty cryptic and the stack traces are pretty hard to parse through

Reply:
Word. That makes sense. I'll work without it! I need readable and testable.